contracts/contracts/vault/VaultCore.sol

```diff
@@ -385,32 +385,102 @@ contract VaultCore is VaultInitializer {
-        if (_trusteeAddress != address(0) && (vaultValue > ousdSupply)) {
-            uint256 yield = vaultValue - ousdSupply;
-            uint256 fee = yield.mulTruncateScale(trusteeFeeBps, 1e4);
-            require(yield > fee, "Fee must not be greater than yield");
-            if (fee > 0) {
-                oUSD.mint(_trusteeAddress, fee);
-            }
-            emit YieldDistribution(_trusteeAddress, yield, fee);
+        uint256 fee = 0;
+        if (_trusteeAddress != address(0)) {
+            fee = (yield * trusteeFeeBps) / 1e4;
+        }
+        if (fee > 0) {
+            require(fee < yield, "Fee must not be greater than yield");
+            oUSD.mint(_trusteeAddress, fee);
         }
+        emit YieldDistribution(_trusteeAddress, yield, fee);
 
         // Only ratchet OToken supply upwards
-        ousdSupply = oUSD.totalSupply(); // Final check should use latest value
-        if (vaultValue > ousdSupply) {
-            oUSD.changeSupply(vaultValue);
+        // Final check uses latest totalSupply
+        uint256 preSupply = oUSD.totalSupply();
+        if (newSupply > preSupply) {
+            oUSD.changeSupply(newSupply);
+            rebasePerSecondTarget = uint64(_min(newTarget, type(uint64).max));
+            lastRebase = uint64(block.timestamp); // Intentional cast
         }
         return vaultValue;
     }
 
+    /**
+     * @notice Calculates the amount that would rebase at at next rebase.
+     * This is before any fees.
+     * @return yield amount of expected yield
+     */
+    function previewYield() external view returns (uint256 yield) {
+        uint256 _totalSupply = oUSD.totalSupply();
+        uint256 _nonRebasing = oUSD.nonRebasingSupply();
+        (yield, ) = _nextYield(_totalSupply, _totalValue(), _nonRebasing);
+        return yield;
+    }
+
+    function _nextYield(
+        uint256 supply,
+        uint256 vaultValue,
+        uint256 nonRebasing
+    ) internal view returns (uint256 yield, uint256 targetRate) {
+        uint256 rebasing = supply - nonRebasing;
+        uint256 elapsed = block.timestamp - lastRebase;
+        targetRate = rebasePerSecondTarget;
+
+        if (
+            elapsed == 0 || // Yield only once per block.
+            rebasing == 0 || // No yield if there are no rebasing tokens to give it to.
+            supply > vaultValue || // No yield if we do not have yield to give.
+            block.timestamp >= type(uint64).max // No yield if we are too far in the future to calculate it correctly.
+        ) {
+            return (0, targetRate);
+        }
+
+        // Start with the full difference available
+        uint256 yield = vaultValue - supply;
+
+        // Cap via optional automatic duration smoothing
+        uint256 _dripDuration = dripDuration;
+        if (_dripDuration > 1) {
+            // If we are able to sustain an increased drip rate for a
+            // longer duration, then increase the target drip rate
```

Comment:
nit: I think this is comment is more accurate
```
// If we are able to sustain an increased drip rate for
// double the duration, then increase the target drip rate
```

Reply:
Changed in 864fa57b881a7ada8acf370a9eddefcc464a9378.

There's actually a bit of range that this number could be. Just ended up on 1/2, which is slightly more capital than is needed, and looks clean. 1.7 actually works out decently, if total supply is constant.